nestkernel/connector_base.h

```diff
@@ -196,10 +197,39 @@ class ConnectorBase
     t_lastspike_ = t_lastspike;
   }
 
+  void send_weight_event( const CommonSynapseProperties& cp,
+    const Event& e,
+    const thread t );
+
 private:
   double t_lastspike_;
 };
 
+inline void
+ConnectorBase::send_weight_event( const CommonSynapseProperties& cp,
+  const Event& e,
+  const thread t )
+{
+  if ( cp.weight_recorders_.size() != 0 )
+  {
+    // Create new event to record the weight and copy relevant content.
+    WeightRecorderEvent wr_e;
+    wr_e.set_port( e.get_port() );
+    wr_e.set_rport( e.get_rport() );
+    wr_e.set_stamp( e.get_stamp() );
+    wr_e.set_sender( e.get_sender() );
+    wr_e.set_sender_gid( e.get_sender_gid() );
+    wr_e.set_weight( e.get_weight() );
+    wr_e.set_delay( e.get_delay() );
+    wr_e.set_receiver(
+      *cp.weight_recorders_[ t ] ); // receiver is the weight recorder
```

Comment:
With `SiblingContainer`, this will turn into `wr_e.set_receiver( *( cp.weight_recorder->get_thread_sibling( t ) ) )`.